### PR TITLE
Add decrypt_string option to ansible-vault

### PR DESCRIPTION
**Pull request by willthames**
##### SUMMARY
Allow encrypted strings inside variable files to be decrypted

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
ansible-vault

##### ANSIBLE VERSION
```
ansible 2.5.0 (devel 83b7ec69fc) last updated 2018/01/12 11:28:16 (GMT +1000)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/will/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/will/src/ansible/lib/ansible
  executable location = /home/will/src/ansible/bin/ansible
  python version = 2.7.14 (default, Dec 11 2017, 14:52:53) [GCC 7.2.1 20170915 (Red Hat 7.2.1-2)]
```

### CODE REVIEW

**Comment by mkrizek** [comment]
cc @alikins

File: docs/docsite/rst/user_guide/vault.rst [hunk 1]
```diff
@@ -194,6 +194,51 @@ Result::
...
+              3866363862363335620a376466656164383032633338306162326639643635663936623939666238
+              3161
+
+Result::
+
+    new_user_password: hunter42
+
+Input::
+
+              $ANSIBLE_VAULT;1.2;AES256;dev
```
Comment (bcoca) [inline docs/docsite/rst/user_guide/vault.rst:230]: isn't this case already covered by decrypt on stdin?

Comment (willthames) [reply docs/docsite/rst/user_guide/vault.rst:230]: It doesn't handle indented vars (which is what you get copying from inventory), but we could fix that in decrypt I suppose

```
$ ansible-vault decrypt
Vault password: 
Reading ciphertext input from stdin
  $ANSIBLE_VAULT;1.1;AES256
  30376...
ERROR! input is not vault encrypted data- is not a vault encrypted file for -
```

**Comment by acozine** [comment]
@willthames are you still interested in this fix? or in handling indented vars some other way?

**Comment by willthames** [comment]
@acozine some form of this fix would be beneficial, but perhaps I'll close this, take @bcoca's suggestion and make `decrypt` handle indented vars.

**Pull request closed without merging**